sui_core/src/authority.rs

```diff
@@ -806,6 +815,145 @@ impl AuthorityState {
         self.database.get_object(object_id)
     }
 
+    pub async fn get_object_info(&self, object_id: &ObjectID) -> Result<ObjectRead, SuiError> {
+        match self.database.get_latest_parent_entry(*object_id)? {
+            None => Ok(ObjectRead::NotExists(*object_id)),
+            Some((obj_ref, _)) => {
+                if obj_ref.2.is_alive() {
+                    match self.database.get_object_version(object_id, obj_ref.1)? {
+                        None => {
+                            error!("Object with in parent_entry is missing from object store, datastore is inconsistent");
+                            Err(SuiError::ObjectNotFound {
+                                object_id: *object_id,
+                            })
+                        }
+                        Some(object) => {
+                            let resolver = ModuleCache::new(&self);
+                            let layout =
+                                object.get_layout(ObjectFormatOptions::default(), &resolver)?;
+                            Ok(ObjectRead::Exists(obj_ref, object, layout))
+                        }
+                    }
+                } else {
+                    Ok(ObjectRead::Deleted(obj_ref))
+                }
+            }
+        }
+    }
+
+    pub async fn get_owned_objects(&self, account_addr: SuiAddress) -> SuiResult<Vec<ObjectRef>> {
+        self.database.get_account_objects(account_addr)
+    }
+
+    pub fn get_total_transaction_number(&self) -> Result<u64, anyhow::Error> {
```

Comment:
From line 848 to 914 are all directly operating on the database. If you move all these functions to the Store, then authority and gateway can share the API and remove redundant code.